### PR TITLE
Settings: Use new language picker in Calypso

### PR DESCRIPTION
**Pull request by sarayourfriend**
#### Changes proposed in this Pull Request

* Use the new language picker in Calypso's language pickers (site language and UI/account language)

#### Testing instructions

* Ensure the language pickers at `/settings/general/:site` and `/me/account` work as expected both on desktop and mobile. Ensure that searching behaves as expected and that selecting a new language works.
* Ensure that checking the empathy mode and fallback checkboxes is working and that they appear appropriately on mobile and desktop (these are only present on the `/me/account` language picker).

### CODE REVIEW

**Comment by sarayourfriend** [comment]
I noticed account balloons by 26% with this PR. Should we async-load the language picker?

**Comment by tyxla** [comment]
> I noticed account balloons by 26% with this PR. Should we async-load the language picker?

Yes, I'm expecting it will be an improvement; and it's a great candidate for that because it isn't immediately necessary.

File: client/components/language-picker/modal.scss [hunk 1]
```diff
@@ -1,204 +1,77 @@
...
-	}
-
-	p {
-		margin: 0;
-	}
-}
+.language-picker__modal-incomplete-locale-tooltip-text {
+    display: flex;
+    flex-direction: column;
+}
```
Comment (tyxla) [inline client/components/language-picker/modal.scss:77]: EOL missing here

**Review by tyxla — APPROVED** [review]
This is looking good to me 👍  Nice work!

@Automattic/i18n - would you like to take a final look before we 🚢  it?

**Review by yuliyan — APPROVED** [review]
I want to drop a few visual nitpicks that I found, though they don't seem to be caused by the changes in this PR. Not sure if those should be addressed here on in a separate PR.

* Regions and languages items are misaligned by a few pixels (~2px), due to the additional padding/border bottom of the region items. Adding padding top to `.language-picker-component__language-group > span` should fix it. As a side effect, it should also fix the off-center focus state of the region items. 
image
image



* Noticed that the region dropdown on mobile gets cut on the right side under certain screen width (~424px). As a solution, maybe the width can be replaced from `85%` to `calc(100% - 50px)`, where 50px is the width of the search button that goes above it.
image


Other than that, I tested the language picker with empathy mode and UI language fallback and everything seems to be working well. LGTM 🚀

**Comment by sarayourfriend** [comment]
Thanks for catching those @yuliyan, I really appreciate your careful review! I've gone ahead and fixed those as part of this PR 🙂

**Comment by yuliyan** [comment]
@saramarcondes, thank you. Looks good!

**Pull request merged**